src/api.js

```diff
@@ -1,23 +1,30 @@
-const api = {
-    doImport (data) {
-        try { 
-            window.application.$store.dispatch('importModel', {
-                clientWidth: document.getElementById('svg-grid').clientWidth,
-                clientHeight: document.getElementById('svg-grid').clientHeight,
-                data: JSON.parse(data)
-            });
-        }catch(err){
-            return false;
-        }
-        return true;
-    },
-
-    doExport () {
-        return window.application.$store.getters['exportData'];
-    },
-
-    setConfig (config) {
-        console.log(config);
-        return true;
+window.api = {
+  config: {},
+  doImport(data) {
+    try {
+      window.application.$store.dispatch('importModel', {
+        clientWidth: document.getElementById('svg-grid').clientWidth,
+        clientHeight: document.getElementById('svg-grid').clientHeight,
+        data: JSON.parse(data),
+      });
+    } catch (err) {
+      return false;
     }
+    return true;
+  },
+  doExport() {
+    return window.application.$store.getters['exportData'];
+  },
+  setConfig({
+    units = 'm',
+    showMapDialogOnStart = true,
+    online = true,
+    onChange = function change() {},
+  }) {
+    this.config = { units, showMapDialogOnStart, online, onChange, };
+
+    startApp();
```

Comment:
It seems odd to me that `setConfig()` would call `startApp()`. not what I would expect as a user of your api. Also, it seems to cause problems if I call `setConfig` more than once:

![](https://trello-attachments.s3.amazonaws.com/58d428743111af1d0a20cf28/596e635ad3346daa47d3a97f/ff2ec674aa8e7c14aa5509f0979a2049/capture.png)